### PR TITLE
Kadanes Algorithm implementation using DART

### PR DESCRIPTION
**Pull request by akashgkrishnan**
# Welcome to Dart community

### **Describe your change:**
Implemented the kadanes algorithm in Dart

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Dart/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [ ] All new Dart files are placed inside an existing directory.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: dynamic_programming/kadanes_algorithm.dart [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import 'dart:math';
+
+int kadanesAlgorithm(List<int> array) {
+  int maxEndingHere = array[0];
+  int maxSoFar = array[0];
+
+  for (int num in array.sublist(1, array.length)) {
+    maxEndingHere = [maxEndingHere + num, num].reduce(max);
+    maxSoFar = [maxSoFar, maxEndingHere].reduce(max);
```
Comment (Parowicz) [inline dynamic_programming/kadanes_algorithm.dart:9]: From what i see you could just 
```suggestion
    maxEndingHere = max(maxEndingHere + num, num);
    maxSoFar = max(maxSoFar, maxEndingHere);
```

**Comment by akashgkrishnan** [comment]
@cclauss sure. I'll be adding in the test cases for the same in some time

File: dynamic_programming/kadanes_algorithm.dart (hunk 1 above)
Comment (akashgkrishnan) [reply dynamic_programming/kadanes_algorithm.dart:9]: @Parowicz  thanks for the suggestion. I'll update with the suggested changes

**Comment by akashgkrishnan** [comment]
@cclauss I have added the test cases as suggested

**Comment by akashgkrishnan** [comment]
@cclauss opened new PR for the same and closed due to some issues

**Pull request closed without merging**